### DISCUSSION TITLE
GraalVM for JDK XX versioning scheme and maven artifacts' versions don't align

### DISCUSSION THREAD

**Discussion by zakkak** [oracle/graal#7556] [Q&A] [answered]
Hi,

The last couple of GraalVM releases are using the GraalVM for JDK XX versioning scheme, but still require users/frameworks to use the old/internal version of GraalVM, e..g for GraalVM for JDK 17 and JDK 20 they are expected to use org.graalvm.sdk:graal-sdk 23.0.x, while for GraalVM for JDK 21 they are expected to use org.graalvm.sdk:graal-sdk (or the new split packages) 23.1.x.

This is already kind of confusing but users can still figure out the right version to use by looking at the JVMCI suffix in the build version of `native-image --version`. Unfortunately it looks like this [will no longer be possible starting with GraalVM for JDK 22](https://github.com/graalvm/labs-openjdk/issues/1).

Are there any thoughts/plans on how to simplify this moving forward?
Switching the maven artifact versions to the new scheme is not expected to work since maven will think that 23.0.x is newer than 21.

Thanks

cc @gilles-duboscq @fniephaus

**Comment by gilles-duboscq** [comment, marked as answer]
Thanks for your question. I agree we need to clarify things in this area.

Some assumptions i think we should make:
* As seen in #6831, we want to make it possible to compile against the newest version of the SDK with as old a JDK as possible.
* As time passes, some features of the SDK will only be in GraalVM for JDK X and above but not below.
* Code compiled against an old SDK version should keep working with new GraalVM versions (modulo a reasonable deprecation timeline).

If it wasn't for the first point above i'd say we can use a 1:1 mapping, but given that we want to be able to compile against the newest version of the SDK with as old a JDK as possible, i think it might be a good to keep a separate version.

As a library maintainer:
* I need to target Java 11 or above
* Then I pick 23.0 or 23.1 depending on the features i need

As an application developer who needs to use the SDK:
* Then I pick 23.0 or 23.1 depending on the Java version i'm using:

| JDK | SDK version train | Latest SDK version |
|--------|--------|--------|
| 17 | 23.0 |  23.0.1 |
| 21 | 23.1 |  23.1.0 |

Regarding the output of `native-image --version`, if we keep doing the above i agree that this "latest supported SDK version" should appear somewhere.
